driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -1593,15 +1593,15 @@ private void connect() {
                 if (serverResponseVersion == attemptedVersion) {
                     // We're post CASSANDRA-11464: the server replies with the same version that the client tried (to
                     // ensure that the client can decode the response).
-                    // Retry with a lower version (if it's too low we'll eventually reach our min and fail).
+                    // Retry with a lower version (if it's too low we'll fail at the next attempt, as we only retry once).
```

Comment:
> what happens if the server supports up to v10 and the client starts with v12?

The server would reply with v10. If the driver also supports v10, then v10 will be used; otherwise, it will fail when decoding the response frame ([here](https://github.com/datastax/java-driver/blob/3.x/driver-core/src/main/java/com/datastax/driver/core/Frame.java#L93-L93) more precisely).